src/librustc_infer/infer/error_reporting/mod.rs

```diff
@@ -1781,6 +1802,30 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
             }
         }
 
+        let new_binding_suggestion =
+            |err: &mut DiagnosticBuilder<'tcx>,
+             type_param_span: Option<(Span, bool, bool)>,
+             bound_kind: GenericKind<'tcx>| {
+                let msg = "consider introducing an explicit lifetime bound to unify the type \
```

Comment:
I'm not crazy about the "unify" wording here, as (a) it feels like jargon but also (b) it doesn't match the meaning of unify that I have ("to make the same"). Perhaps "to connect" or something..but see my notes below on the suggestion anyhow.

Reply:
Please let me know what you think of the new wording: "consider introducing an explicit lifetime bound". It seems less normative/explicit, but works well enough in context, IMO.

Reply:
I prefer that, yes.